core/src/components/alert/alert.ios.scss

```diff
@@ -105,6 +105,10 @@
   &::-ms-clear {
     display: none;
   }
+
+  &::-webkit-date-and-time-value {
```

Comment:
This pseudo selector only appears on Mobile Safari. For Safari it is `::-webkit-date-and-time`. So this fix is specifically targeted for the problem we face on Mobile Safari.

Reply:
If the problem is that `appearance` is being set to `none`, would it make sense to set `appearance: auto` instead to counteract it without needing to set a specific height?

Either way, can we add a comment explaining why the change is needed, and why we didn't use `::webkit-date-and-time`?

Reply:
[`appearance: auto`](https://developer.mozilla.org/en-US/docs/Web/CSS/appearance#auto) would act the same as removing the explicit `appearance` style and letting the vendor-specific styling take effect. 

On iOS an input of `type="date"` has explicit width and height styles on the pseudo element:
```css
height: 18px;
width: 86px;
```

Would a comment saying that the pseudo selector comes from iOS be sufficient? I don't know how else to describe why we didn't use `-webkit-date-and-time` other than it isn't the pseudo selector rendered:

![CleanShot 2023-11-13 at 12 48 10](https://github.com/ionic-team/ionic-framework/assets/13732623/ba7373ad-fed3-41b5-866d-3bcec7f3d7fe)

Reply:
I would include that (1) we need to set an explicit height to fix mobile Safari squishing the component down and (2) the pseudo-selector is specific to mobile Safari, similar to the original PR comment you put here.